dashboard/src/containers/RoutesContainer/Routes.test.tsx

```diff
@@ -77,3 +78,51 @@ it("should render a redirect to the login page (when not authenticated)", () =>
       .render().props.to,
   ).toEqual("/login");
 });
+
+describe("Routes depending on feature flags", () => {
+  const namespace = "default";
+  const perNamespacePath = "/config/ns/:namespace/repos";
+  const nonNamespacedPath = "/config/repos";
+
+  it("should use a non-namespaced route for app repos without feature flag", () => {
+    const wrapper = shallow(
+      <StaticRouter location="/config/repos" context={{}}>
+        <Routes
+          {...emptyRouteComponentProps}
+          namespace={namespace}
+          authenticated={true}
+          featureFlags={{ reposPerNamespace: false }}
+        />
+      </StaticRouter>,
+    )
+      .dive()
+      .dive()
+      .dive();
```

Comment:
what's the `dive()` for? I guess it's to be able to find the component below but didn't `mount(<StaticRouter ...`  worked for you?

Reply:
Because `mount` renders the component and all children, it was also rendering the `PrivateRouteContainer` which requires a connected store. `shallow`, where possible, seems to be preferred for both this reason and the fact that you can focus your test on the component level unit-test, rather than forcing to test all the children too (ie. this test won't fail if `RepoListContainer` has an issue, only `RepoListContainer`'s unit tests will).

It does look a bit ugly with the three `dive()`'s though, to get to the `RepoListContainer` only.